src/chrome/chromeDebugAdapter.ts

```diff
@@ -472,6 +481,60 @@ export abstract class ChromeDebugAdapter implements IDebugAdapter {
         } else if (notification.hitBreakpoints && notification.hitBreakpoints.length) {
             reason = 'breakpoint';
 
+            const hitBreakpoints = notification.hitBreakpoints;
+            let allStopOnEntryBreakpoints = true;
+
+            // If there is a breakpoint which is not a stopOnEntry breakpoint, we appear as if we hit that one
+            // This is particularly done for cases when we end up with a user breakpoint and a stopOnEntry breakpoint on the same line
+            hitBreakpoints.forEach(bp => {
+                if (!this._stopOnEntryBreakpointIdToRequestedFileName.has(bp)) {
+                    notification.hitBreakpoints = [bp];
+                    allStopOnEntryBreakpoints = false;
+                }
+            });
+
+            // If all the breakpoints on this point are stopOnEntry breakpoints
+            // This will be true in cases where it's a single breakpoint and it's a stopOnEntry breakpoint
+            // This can also be true when we have multiple breakpoints and all of them are stopOnEntry breakpoints, for example in cases like index.js and index.bin.js
```

Comment:
Suppose user puts breakpoints in both index.js and index.bin.js files, when the setBreakpoints function is called for index.js it will set a stopOnEntry breakpoint on *index.* files which will also match index.bin.js. 
Now when setBreakpoints is called for index.bin.js it will again put a stopOnEntry breakpoint in itself. So when the file is actually loaded, we would have 2 stopOnEntry breakpoints.